velox/common/base/PeriodicStatsReporter.cpp

```diff
@@ -63,7 +63,11 @@ PeriodicStatsReporter::PeriodicStatsReporter(const Options& options)
       cache_(options.cache),
       arbitrator_(options.arbitrator),
       spillMemoryPool_(options.spillMemoryPool),
-      options_(options) {}
+#ifdef VELOX_ENABLE_S3
+      s3Metrics(options.s3Metrics),
+#endif
+      options_(options) {
+}
 
 void PeriodicStatsReporter::start() {
   LOG(INFO) << "Starting PeriodicStatsReporter with options "
```

Comment:
There are too many #ifdef VELOX_ENABLE_S3.. We need to make S3MetricsAggregator register with PeriodicStatsReporter. May be rewrite PeriodicStatsReporter to register a task.

Reply:
I have refactored `PeriodicStatsReporter` to register the `S3MetricsAggregator` as a periodic task. Now, the S3 metrics are registered using the new method `addS3MetricsAggregatorTask()`, which is called when `VELOX_ENABLE_S3` is defined and `s3Metrics` is provided. This refactor reduces the number of `#ifdef VELOX_ENABLE_S3` conditions.